_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -378,7 +362,7 @@ public static function get_user_connection_data_permission_callback() {
 	 * @return bool|WP_Error True if user is able to unlink.
 	 */
 	public static function unlink_user_permission_callback() {
-		if ( current_user_can( 'jetpack_connect' ) || Jetpack::is_user_connected( get_current_user_id() ) ) {
+		if ( current_user_can( 'jetpack_connect_user' ) && Jetpack::is_user_connected( get_current_user_id() ) ) {
```

Comment:
Changes made.  Callback nor route accepts any ID param, and the endpoint will only disconnect current user as per `Jetpack::unlink_user()`